structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/controller/CDPStructuredEventV1Controller.java

```diff
@@ -55,4 +71,34 @@ private Response getAuditEventsZipResponse(Collection<CDPStructuredNotificationE
         String fileName = String.format("audit-%s.zip", resourceType);
         return Response.ok(streamingOutput).header("content-disposition", String.format("attachment; filename = %s", fileName)).build();
     }
+
+    private void checkPermissions(Collection<? extends CDPStructuredEvent> events) {
+        for (CDPStructuredEvent event : events) {
+            String resourceType = event.getOperation().getResourceType();
+            Arrays.asList(AuthorizationResourceAction.values()).stream()
+                    .filter(action -> isResourceTypeHasDescribeOrGetAction(action, resourceType))
+                    .findFirst()
+                    .ifPresentOrElse(
+                            action -> checkPermissionForResource(action, event),
+                            () -> throwIllegalStateExceptionForResourceType(resourceType)
+                    );
+        }
+    }
+
+    private boolean isResourceTypeHasDescribeOrGetAction(AuthorizationResourceAction action, String resourceType) {
+        return (action.name().startsWith(DESCRIBE_ACTION_PREFIX) || action.name().contains(GET_ACTION_PREFIX))
+                && action.name().contains(resourceType.toUpperCase());
+    }
+
+    private void checkPermissionForResource(AuthorizationResourceAction action, CDPStructuredEvent event) {
+        permissionCheckingUtils.checkPermissionForUserOnResource(
+                action,
+                ThreadBasedUserCrnProvider.getUserCrn(),
+                event.getOperation().getResourceCrn());
+    }
+
+    private void throwIllegalStateExceptionForResourceType(String resourceType) {
+        throw new IllegalStateException("Unable to find AuthZ action for resource: " + resourceType);
+    }
```

Comment:
the logic here and the logic in AuditController seems to have a lot in common, so I suggest to create a common logic for them, something like `EventAuthorizationUtils`